extra_data/read_machinery.py

```diff
@@ -271,3 +271,22 @@ def same_run(*args) -> bool:
 
 
 glob_wildcards_re = re.compile(r'([*?[])')
+
+
+# Based on _alloc function in pasha
+def zeros_shared(shape, dtype):
```

Comment:
Could we expose this? Or add something like `KeyData.allocate_out(shared=True)`?